sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -210,6 +197,17 @@ case class DataSourceResolution(
           }
           ShowTablesCommand(Some(namespace.quoted), pattern)
       }
+
+    case UseCatalogAndNamespaceStatement(catalogName, namespace) =>
+      catalogName match {
+        case Some(c) =>
+          validateCatalog(c)
+          UseCatalogAndNamespace(catalogManager, Some(c), namespace)
+        case None =>
+          require(namespace.nonEmpty)
+          val CurrentCatalogAndNamespace(catalog, ns) = namespace.get
+          UseCatalogAndNamespace(catalogManager, Some(catalog.name()), Some(ns))
```

Comment:
Actually, this may not be correct if the catalog is `V2SessionCatalog`. The V1 behavior is that it calls `SessionCatalog.setCurrentDatabase(databaseName)`. So if we don't fallback to V1, any code that access `SessionCatalog.getCurrentDatabase` can be broken.

Reply:
Fixed.